pkg/proxy/nftables/proxier.go

```diff
@@ -0,0 +1,1542 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nftables
+
+//
+// NOTE: this needs to be tested in e2e since it uses iptables for everything.
+//
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base32"
+	"fmt"
+	"net"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/danwinship/nftables"
+
+	v1 "k8s.io/api/core/v1"
+	discovery "k8s.io/api/discovery/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/events"
+	utilsysctl "k8s.io/component-helpers/node/util/sysctl"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/proxy"
+	"k8s.io/kubernetes/pkg/proxy/conntrack"
+	"k8s.io/kubernetes/pkg/proxy/healthcheck"
+	"k8s.io/kubernetes/pkg/proxy/metaproxier"
+	"k8s.io/kubernetes/pkg/proxy/metrics"
+	proxyutil "k8s.io/kubernetes/pkg/proxy/util"
+	proxyutiliptables "k8s.io/kubernetes/pkg/proxy/util/iptables"
+	"k8s.io/kubernetes/pkg/util/async"
+	utilexec "k8s.io/utils/exec"
+	netutils "k8s.io/utils/net"
+)
+
+const (
+	// the nftables table
```

Comment:
I just noticed we move from uppercase to lowercase on the chains names, is that for some specific reason?

Reply:
iptables and nftables both allow either uppercase or lowercase, but uppercase is the usual convention with iptables and lowercase with nftables... :man_shrugging: 

(We could also do something like "standard chain names are lowercase and per-service/endpoint chain names are uppercase" or vice versa...)

Reply:
> iptables and nftables both allow either uppercase or lowercase, but uppercase is the usual convention with iptables and lowercase with nftables... 🤷‍♂️

or rather: iptables and nftables both allow either uppercase or lowercase letters, and hyphens and underscores, but uppercase-with-hyphens is the usual convention with iptables, and lowercase-with-underscores is the usual convention with nftables.

(And I'm using lowercase-with-hyphens... :scream:.)

Reply:
Why lower-with-hyphens if underscores are normal?

Reply:
I don't really have a good answer here. One part of my brain was saying "you should use lowercase because that's normal for nftables" and another part of my brain was saying "you should use hyphens because that's normal for kube-proxy".

I'd be fine changing it, though there's nothing actually _wrong_ with how it is now. It's not going to end up breaking some tools or anything like that, it just doesn't match other people's naming conventions.